components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -20,23 +20,36 @@ namespace scream
 // according to the "averaging" type, and according to the number of
 // model time steps since the last output step.
 KOKKOS_INLINE_FUNCTION
-void combine (const Real& new_val, Real& curr_val, const OutputAvgType avg_type)
+void combine (const Real& new_val, Real& curr_val, Real& avg_coeff, const OutputAvgType avg_type, const Real fill_value)
 {
-  switch (avg_type) {
-    case OutputAvgType::Instant:
-      curr_val = new_val;
-      break;
-    case OutputAvgType::Max:
-      curr_val = ekat::impl::max(curr_val,new_val);
-      break;
-    case OutputAvgType::Min:
-      curr_val = ekat::impl::min(curr_val,new_val);
-      break;
-    case OutputAvgType::Average:
-      curr_val += new_val;
-      break;
-    default:
-      EKAT_KERNEL_ERROR_MSG ("Unexpected value for m_avg_type. Please, contact developers.\n");
+  const bool new_fill  = new_val == fill_value;
+  const bool curr_fill = curr_val == fill_value;
+  if (!new_fill) {
```

Comment:
There are lots of conditionals here. For "normal" output, all of these are checked, which may slow down things, since the conditions may change across iterations (except for the avg type switch). Can we attach a flag to the fields that "may" contain fill values, and only do all these checks for those? Perhaps we should have `combine` and `combine_with_fill`.

Beside performance, my real concern is that all these cases of fill values are making our code quite complex and hard to follow, in a lot of places, all for the sake (if I understand correctly) of one particular use case (vertical remap) which we don't use that often.

I know we all have projects to follow, and deadlines, so I won't make a stand, but I'm going to open an issue, and see if we can clean up a bit.